### PR TITLE
Fix testIndexAdd on GPU

### PR DESCRIPTION
**Pull request by JackCaoG**
Same as https://github.com/pytorch/xla/pull/2205, I messed up the submodule id on that one.

IndexAdd failure on XLA:GPU is related to the precision, change to AllClose which has default rtol set to 1e-5

### CODE REVIEW

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -4691,19 +4691,17 @@ TEST_F(AtenXlaTensorTest, TestIndexAdd) {
...
+        torch::Tensor xla_value = CopyToDevice(value, device);
+        torch::Tensor xla_result =
+            torch::index_add(xla_base, dim, xla_index, xla_value);
+        AllClose(result, xla_result);
 
-            ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
-            ExpectCounterChanged("xla::index_add_",
-                                 cpp_test::GetIgnoredCounters());
-          });
+        ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
```
Comment (dlibenzi) [inline test/cpp/test_aten_xla_tensor.cpp:4702]: Move those before returning from function. Same other places you touched.

Comment (JackCaoG) [reply test/cpp/test_aten_xla_tensor.cpp:4702]: Sorry I am a bit confuse. ExpectCounterNotChanged is still inside the `ForEachDevice` scope. The changes I made was replacing the `AllEqual` with `AllClose` and delete the `{DeviceType::CPU, DeviceType::TPU}` so GPU will also run this test. The rest was formatting tool adjusting the indent.

Comment (dlibenzi) [reply test/cpp/test_aten_xla_tensor.cpp:4702]: These calls should be before the return.
You can leave them there and I will change them in another PR.

Comment (JackCaoG) [reply test/cpp/test_aten_xla_tensor.cpp:4702]: Oh, I see your point. yea I will make the change

**Pull request merged**